### PR TITLE
phoenix_live_view.js: Regex simplification

### PR DESCRIPTION
**Pull request by henrik**
I don't know the incantation to update priv/static/phoenix_live_view.js from this one.

### CODE REVIEW

File: assets/js/phoenix_live_view.js [hunk 1]
```diff
@@ -556,7 +556,7 @@ export class LiveSocket {
   }
 
   withinTargets(el, phxTarget, callback){
-    if(/^(0|[1-9](\d?)+)$/.test(phxTarget)){
+    if(/^(0|[1-9]\d*)$/.test(phxTarget)){
```
Comment (sanghun89) [inline assets/js/phoenix_live_view.js:559]: The resource here is pretty cheap, but we can save re-instantiating RegExp objects on every check by extracting this as a separate static member of this class.

**Comment by henrik** [comment]
Ah, this incantation seems to work:

```
cd assets
npx webpack --mode production
git diff --word-diff  # to confirm the change
```

I force-pushed with an updated `priv/static/phoenix_live_view.js` – not sure if that's the way it should be done.

File: assets/js/phoenix_live_view.js (hunk 1 above)
Comment (henrik) [reply assets/js/phoenix_live_view.js:559]: I think I'll skip that in this PR to keep it focused to one thing, but fair point :)

**Comment by henrik** [comment]
Force pushed again to remove the update of the minified JS to avoid any merge issues. It should probably be updated post-merge as mentioned in #975.

**Comment by chrismccord** [comment]
Can you resolve conflicts with master? Thanks!

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Comment by henrik** [comment]
Thank you, @chrismccord! :)

**Pull request merged**